esphome/components/shelly_dimmer/light.py

```diff
@@ -75,22 +174,26 @@
             cv.Optional(CONF_GAMMA_CORRECT, default=1.0): cv.positive_float,
         }
     )
-    .extend(cv.COMPONENT_SCHEMA)
+    .extend(cv.polling_component_schema("10s"))
     .extend(uart.UART_DEVICE_SCHEMA)
 )
 
 
 def to_code(config):
-    fw_major, fw_minor, fw_file = config[CONF_FIRMWARE].enum_value
-    firmware_path = (Path(__file__).parent / fw_file).resolve()
-    firmware_data = Path(firmware_path).read_bytes()
-    hexdata = [HexInt(x) for x in firmware_data]
-    cg.add_define("SHD_FIRMWARE_DATA", hexdata)
+    fw_hex = get_firmware(config[CONF_FIRMWARE])
+    fw_major, fw_minor = parse_firmware_version(config[CONF_FIRMWARE][CONF_VERSION])
+
+    if fw_hex is not None:
+        cg.add_define("SHD_FIRMWARE_DATA", fw_hex)
     cg.add_define("SHD_FIRMWARE_MAJOR_VERSION", fw_major)
     cg.add_define("SHD_FIRMWARE_MINOR_VERSION", fw_minor)
```

Comment:
Please prefix these with `USE_` to match other defines we use in ESPHome
```suggestion
        cg.add_define("USE_SHD_FIRMWARE_DATA", fw_hex)
    cg.add_define("USE_SHD_FIRMWARE_MAJOR_VERSION", fw_major)
    cg.add_define("USE_SHD_FIRMWARE_MINOR_VERSION", fw_minor)
```